reference-implementation/to-upstream-wpts/writable-streams/brand-checks.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+}
+
+function fakeWritableStreamDefaultWriter() {
+  return {
+    get closed() { return Promise.resolve(); },
+    get desiredSize() { return 1; },
+    get ready() { return Promise.resolve(); },
+    abort() { return Promise.resolve(); },
+    close() { return Promise.resolve(); },
+    write() { return Promise.resolve(); }
+  };
+}
+
+function realReadableStreamDefaultWriter() {
+  const rs = new ReadableStream();
+  return rs.getReader();
+}
+
+function getterRejects(t, obj, getterName, target) {
+  const getter = Object.getOwnPropertyDescriptor(obj, getterName).get;
+
+  return promise_rejects(t, new TypeError(), getter.call(target),
+    getterName + ' should reject with a TypeError');
+}
+
+function methodRejects(t, obj, methodName, target) {
+  const method = obj[methodName];
+
+  return promise_rejects(t, new TypeError(), method.call(target),
+    methodName + ' should reject with a TypeError');
+}
+
+function getterThrows(obj, getterName, target) {
+  const getter = Object.getOwnPropertyDescriptor(obj, getterName).get;
+
+  assert_throws(new TypeError(), () => getter.call(target), getterName + ' should throw a TypeError');
+}
+
+const ws = new WritableStream();
+const writer = ws.getWriter();
+const WritableStreamDefaultWriter = writer.constructor;
+const WriterProto = WritableStreamDefaultWriter.prototype;
+
+test(() => {
+  getterThrows(WriterProto, 'desiredSize', fakeWritableStreamDefaultWriter());
+  getterThrows(WriterProto, 'desiredSize', realReadableStreamDefaultWriter());
+}, 'WritableStreamDefaultWriter.prototype.desiredSize enforces a brand check');
+
+promise_test(t => {
+  const results = [];
```

Comment:
Sure. That works for this one, but some tests used a for loop so that doesn't work for them without a bigger rework.